controllers/mysqlcluster_controller.go

```diff
@@ -1862,6 +1875,71 @@ func (r *MySQLClusterReconciler) finalizeV1(ctx context.Context, cluster *mocov1
 	return nil
 }
 
+func (r *MySQLClusterReconciler) updateStatusByStatefulSet(ctx context.Context, cluster *mocov1beta2.MySQLCluster) error {
+	log := crlog.FromContext(ctx)
+
+	var sts appsv1.StatefulSet
+	err := r.Get(ctx, client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.PrefixedName()}, &sts)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get StatefulSet %s/%s: %w", cluster.Namespace, cluster.PrefixedName(), err)
+	}
```

Comment:
If the err is `NotFound`, it sets `StatefulSetReady` false because the condition is `err != nil && !apierrors.IsNotFound(err)`
When only err is not `NotFound`, it returns error.

Reply:
Oh, I was mistaken. Thank you for the reply.

Reply:
I found that this implementation is not good because if statufulset does not found, the later process would no t be done due to early return.
So I will change the implementation.

Reply:
Thank you for adding a new test case.

BTW, Please divide into cases that the error returned by `Get()` is NotFound or other error.
In very rare cases, an error other than NotFound may be returned.